--- /usr/lib64/python2.6/subprocess.py	2014-09-01 13:18:47.000000000 +0800
+++ subprocess2.6_new.py	2014-09-01 13:17:52.000000000 +0800
@@ -1168,7 +1168,11 @@
 
                             if cwd is not None:
                                 os.chdir(cwd)
-
+			    import signal
+                            signals = ('SIGPIPE', 'SIGXFZ', 'SIGXFSZ')
+                            for sig in signals:
+                                if hasattr(signal, sig):
+                                    signal.signal(getattr(signal, sig), signal.SIG_DFL)
                             if preexec_fn:
                                 preexec_fn()
 
