--- /usr/lib64/python2.7/subprocess.py	2012-07-24 18:05:38.000000000 +0800
+++ subprocessFix.py	2014-08-13 14:43:34.547954714 +0800
@@ -438,7 +438,7 @@
 
 
 __all__ = ["Popen", "PIPE", "STDOUT", "call", "check_call",
-           "check_output", "CalledProcessError"]
+           "check_output", "CalledProcessError", "_eintr_retry_call"]
 
 if mswindows:
     from _subprocess import (CREATE_NEW_CONSOLE, CREATE_NEW_PROCESS_GROUP,
@@ -1193,6 +1193,11 @@
 
                             if cwd is not None:
                                 os.chdir(cwd)
+			    import signal
+                            signals = ('SIGPIPE', 'SIGXFZ', 'SIGXFSZ')
+                            for sig in signals:
+                                if hasattr(signal, sig):
+                                    signal.signal(getattr(signal, sig), signal.SIG_DFL)
 
                             if preexec_fn:
                                 preexec_fn()
